packages/elements/src/datetime-picker/__test__/datetime-picker.value.test.js

```diff
@@ -246,5 +246,109 @@ describe('datetime-picker/Value', function () {
+      await elementUpdated(el);
+      expect(el.error).to.be.equal(false, 'error state should be false when both values are valid');
+
+      el.values = ['2024-03-01', '2024-03-03'];
+      await elementUpdated(el);
+      expect(el.error).to.be.equal(true, 'error state should be true when input from value is invalid');
+
+      el.values = ['2024-03-02', '2024-03-03'];
+      await elementUpdated(el);
+      expect(el.error).to.be.equal(
+        false,
+        'error state should return to false when input from value is valid'
+      );
+
+      el.values = ['2024-03-02', '2024-03-04'];
+      await elementUpdated(el);
+      expect(el.error).to.be.equal(true, 'error state should be true when input to value is invalid');
+
+      el.values = ['2024-03-02', '2024-03-03'];
+      await elementUpdated(el);
+      expect(el.error).to.be.equal(false, 'error state should return to false when input to value is valid');
+
+      el.values = ['2024-03-04', '2024-03-05'];
+      await elementUpdated(el);
+      expect(el.error).to.be.equal(true, 'error state should be true when both inputs value are invalid');
+
+      el.values = ['2024-03-02', '2024-03-03'];
+      await elementUpdated(el);
+      expect(el.error).to.be.equal(
+        false,
+        'error state should return to false when both inputs value are valid'
+      );
+    });
+    it('It should error when weekdays-only attribute is set and values are not in weekday period with range mode', async function () {
+      const el = await fixture('<ef-datetime-picker weekdays-only lang="en-gb" opened></ef-datetime-picker>');
+
+      el.values = ['2024-03-04', '2024-03-08'];
+      await elementUpdated(el);
+      expect(el.error).to.be.equal(false, 'error state should be false when both values are valid');
+
+      el.values = ['2024-03-03', '2024-03-08'];
+      await elementUpdated(el);
+      expect(el.error).to.be.equal(true, 'error state should be true when input from value is invalid');
+
+      el.values = ['2024-03-04', '2024-03-08'];
+      await elementUpdated(el);
+      expect(el.error).to.be.equal(
+        false,
+        'error state should return to false when input from value is valid'
+      );
+
+      el.values = ['2024-03-04', '2024-03-09'];
+      await elementUpdated(el);
+      expect(el.error).to.be.equal(true, 'error state should be true when input to value is invalid');
+
+      el.values = ['2024-03-04', '2024-03-08'];
+      await elementUpdated(el);
+      expect(el.error).to.be.equal(false, 'error state should return to false when input to value is valid');
+
+      el.values = ['2024-03-03', '2024-03-09'];
+      await elementUpdated(el);
+      expect(el.error).to.be.equal(true, 'error state should be true when both inputs value are invalid');
+
+      el.values = ['2024-03-04', '2024-03-08'];
+      await elementUpdated(el);
+      expect(el.error).to.be.equal(
+        false,
+        'error state should return to false when both inputs value are valid'
```

Comment:
Shouldn't do multiple expect in one test case. And this Sound like multiple test cases.